src/models/basecog.py

```diff
@@ -1,8 +1,9 @@
 from nextcord.ext import commands
+from nextcord import Interaction, Client
 
 
 class BaseCog(commands.Cog):
-    def __init__(self, bot):
+    def __init__(self, bot: Client):
```

Comment:
what exactly is the reason for typehinting with `Client`?

Reply:
looking at the code, it seemed to produce the most similer typhinting

Reply:
can you link some snippet?

Reply:
https://github.dev/nextcord/nextcord/blob/c1c9921cf4a302d51d9bb4657892efa7f7d4ca46/nextcord/ext/commands/bot.py#L1217

I seem to have found the correct class, but the `commands.Bot` seemes to be more adequate, will change accordingly.

Reply:
https://github.com/nextcord/nextcord/blob/c1c9921cf4a302d51d9bb4657892efa7f7d4ca46/nextcord/ext/commands/bot.py#L1217 that technique used in that line wasn't typehinting.

Reply:
it was extending the class of `nextcord.Client` thus having those typehinting of client, but prob lookes better.

> This class is a subclass of :class:`nextcord.Client` and as a result
    anything that you can do with a :class:`nextcord.Client` you can do with
    this bot.

Reply:
ic. still fine because typehinting does nothing at runtime. like you can do `a: int = "1"` and the code still works